Update manifest on push to main

carlreid opened the pull request:
This simply executes the `helm template` command and then performs a diff with git to see if there were any changes. If there are changes, then it will create a commit with those changes with the message `[CI] Syncing Helm manifest`.

I did initially have this run on a PR trigger, but it's very annoying if you have made changes to the chart since then you'd need to pull those changes down. Then if you were to follow up with more changes to the chart, it would result in another automated commit. Probably would add quite some noise to the commit history of `[CI] Syncing Helm manifest` 😄 

> The only "problem" I have found with this approach, is that a lot of tags are added to the manifests that are completely unnecessary and pollute them  

In response to this, I assume this is in relation to `helm.sh/chart` and `app.kubernetes.io/managed-by`. I'd disagree that they're unnecessary as the `app.kubernetes.io/managed-by` is a [recommended label](https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/) and so is the `helm.sh/chart` as stated [here](https://helm.sh/docs/chart_best_practices/labels/#standard-labels). So I think it's a good thing that they're being included? Maybe there's good reasons you guys don't want them in though?

[Example run of the action](https://github.com/carlreid/meilisearch-kubernetes/runs/2160300551?check_suite_focus=true).
[Example of the automated commit](https://github.com/carlreid/meilisearch-kubernetes/commit/8cab5c530dcaafce3d6790a6adf6b07737bd5d5c) (can also check the [commit history](https://github.com/carlreid/meilisearch-kubernetes/commits/main))
Resolves: #12

eskombro reviewed (commented):
Hi again @carlreid !

Thanks a lot for this PR! This approach is very interesting, it let us get rid of the need to update the manifests (and check if they are consistent), since the CI will do the job, and that is amazing! I agree with your approach when it comes to triggering the CI only when a push has been done on `main`, and even more if it is modifying the manifests only when a diff is detected. 🎉 

> In response to this, I assume this is in relation to `helm.sh/chart` and `app.kubernetes.io/managed-by`. I'd disagree that they're unnecessary as the `app.kubernetes.io/managed-by` is a [recommended label](https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/) and so is the `helm.sh/chart` as stated [here](https://helm.sh/docs/chart_best_practices/labels/#standard-labels). So I think it's a good thing that they're being included? Maybe there's good reasons you guys don't want them in though?

I was making reference indeed to those tags! I agree with what you said about these tags being recommended/useful, but that is only in a Helm package context. But from my POV the goal of having manifests is to be able to use them directly, without Helm (or depending on it). These manifests should be able to be applied directly on a k8s cluster, and this is why I feel that in this case, those tags are polluting. If I just use the `meilisearch.yaml` file and apply it to my k8s cluster, I will be adding tags like `app.kubernetes.io/managed-by: Helm` or `helm.sh/chart: meilisearch-X.X.X` that aren't really pertinent anymore, since I haven't even used Helm. WDYT? Do you think there is a way to get rid of them? Or do you still think that they are useful?

carlreid commented:
> the goal of having manifests is to be able to use them directly, without Helm (or depending on it). These manifests should be able to be applied directly on a k8s cluster, and this is why I feel that in this case, those tags are polluting.

I did think after, why you may want them and that's the conclusion I came to. Anyhow, I agree that they're really odd to have if you didn't use Helm to actually deploy it. I've updated the PR now and introduced a `grep` between the template and output file that will skip over lines that include `helm.sh/chart:` or `app.kubernetes.io/managed-by:` before writing the output. You can see the updated result [here](https://github.com/carlreid/meilisearch-kubernetes/commit/31c965ce829d3c70f466f9cbed511d94fa437c8b).

eskombro commented:
That is perfect @carlreid !!!

Amazing, proceeding to merge, thanks a lot!

bors try

eskombro reviewed (approved):
🎉 🎉 🎉

eskombro commented:
Bors merge

carlreid commented:
`You're not authorized to push to this branch`

@eskombro Hmm... You guys may have to help out here. Guess since you have bors managing the merges then it would need write permissions.

eskombro commented:
Oh. I forgot about this 'little' detail: `main` is a protected branch and the CI is [not able to push on it](https://github.com/meilisearch/meilisearch-kubernetes/runs/2187641165?check_suite_focus=true) with the GH token. This is complex, since allowing push on main with GH token would allow any user with write access to push on main, which makes the branch protection useless...

Any idea on how to find a workaround for this problem?

Will discuss this with @curquiza tomorrow, since she handles the rights on the repo!
